src/Sylius/Bundle/ShopBundle/Twig/Component/Checkout/Address/FormComponent.php

```diff
@@ -86,7 +100,14 @@ public function addressFieldUpdated(#[LiveArg] mixed $addressId, #[LiveArg] stri
         $newAddress['city'] = $address->getCity();
         $newAddress['postcode'] = $address->getPostcode();
 
-        $this->formValues[$field] = $newAddress;
+        $formData = new ArrayObject($newAddress);
+
+        $this->eventDispatcher?->dispatch(
+            new CheckoutAddressUpdatedEvent($formData, $address),
+            ShopEvents::CHECKOUT_ADDRESS_UPDATED,
+        );
```

Comment:
Instead of introducing a new event here, I’d suggest adding an `AddressFormValuesModifierInterface` injected as a tagged iterator. This way the component exposes a clear and discoverable extension point: predictable order (via priority), easier debugging, and unit-testability. Events are more implicit and harder to track, while Sylius’ convention is usually small, single-responsibility services wired via DI. This gives plugin authors a first-class, BC-friendly hook without relying on vendor code internals.